src/run.rs

```diff
@@ -55,19 +56,27 @@ pub fn generate_tmp_project(
 ) -> Result<TempDir> {
     let tmp_dir = {
         let mut builder = tempfile::Builder::new();
+        let mut suffix = String::new();
 
-        if cli.worktree_branch.is_some() {
-            builder.prefix("wk-");
+        let prefix = if cli.worktree_branch.is_some() {
+            "wk-"
         } else {
-            builder.prefix("tmp-");
-        }
+            "tmp-"
+        };
+
+        if let Some(name) = cli.project_name.as_deref() {
+            suffix = format!("-{}", name);
+        };
 
         if !temporary_project_dir.exists() {
             fs::create_dir_all(temporary_project_dir)
                 .context("cannot create temporary project's directory")?;
         }
 
-        builder.tempdir_in(temporary_project_dir)?
+        builder
+            .prefix(&prefix)
+            .suffix(&suffix)
```

Comment:
You really are fighting the builder pattern to have it your way.

Try to call `.suffix()` conditionally to avoid the extra variable.

(The variable prefix can also be entirely removed)

Reply:
My problem here is when I create the suffix
```rust
if let Some(name) = cli.project_name.as_deref() {
    builder.suffix(format!("-{}", name).as_str());
}
```
I need a reference and then I start fighting with the borrow checker because the value is freed too early.

Reply:
Most unfortunate :thinking: ...

You're right, there isn't any other way. Why make a builder if the user needs to keep ownership of everything they pass... maybe worth asking on tempfile's project. It's probably fixable using Cow